File: cpp_e9zMZ/cpp/06-thaphinh/85.cpp
#include <iostream>

#include "zvn.h"
using namespace std;

int main() {
    int height = 7;
    string shape = "";
    string space = " ";

    for (int i = 1; i <= height; i++) {
        string record = "";
        for (int j = 1; j <= height; j++) {
            if (j == i || j == height - i + 1) {
                record += to_string(j);
            } else {
                record += space;
            }
        }
        shape += record + "\n";
    }

    cout << shape << endl;
}

File: python_ToSXs/python/06-thaphinh/85.py
height = 7
shape = ""
shapeTop = ""
shapeBottom = ""
center = int(height / 2)

line = "1" + " " * (height - 2) + str(height)
for i in range(2, center + 1):
    record = (" " * (i - 1) +
              str(i) +
              " " * (height - 2 * i) +
              str(height - i + 1))
    shapeTop += record + "\n"

centerCharacter = " " * center + str(center + 1) + "\n"
for i in range(2 + center, height):
    record = (" " * (height - i) +
              str(height - i + 1) +
              " " * (2 * i - height - 2) +
              str(i))
    shapeBottom += record + "\n"

shape = line + "\n" + shapeTop + centerCharacter + shapeBottom + line
print(shape)
